messaging/WFLY-6660_artemis_data_tools.adoc

```diff
@@ -0,0 +1,89 @@
+This new operation would print content of journal to a file. It would take advantage of https://activemq.apache.org/components/artemis/documentation/latest/data-tools.html[Artemis data tools print command].
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-6660[WFLY-6660]
+
+=== Related Issues
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+The new operation 'data-print' will added  to the https://wildscribe.github.io/WildFly/21.0/subsystem/messaging-activemq/server/[/subsystem=messaging-activemq/server resource]. It will print a report about journal records and summary of existent records, as well a report on paging. This operation shouldn't be executed against a running broker.
+
+=== Hard Requirements
+
+Provide a new 'data-print' operation that will return a stream with the following parameters:
+ * safe a Boolean parameter with a default value of 'false' to print the data structure without showing the your data.
+
+=== Nice-to-Have Requirements
+
+ Add an `archive` Boolean parameter to return a zip of the journal dump instead of the plain text report to reduce network usage.
+
+=== Non-Requirements
+
+This operation will only work on the embedded borker and is not by any mean a way to get data from another Artemis installation (standalone or embeded) using the configuration of the broker it is run against.
+Support for JDBC journal is not a requirement as there is no JDBC connection available on admin-mode.
+
+== Implementation Plan
+
+Provide a new RuntimeOnlyHandler that will execute only in `admin-mode`. It will work in the 'same' way as the `org.wildfly.extension.messaging.activemq.ExportJournalOperation` to invoke the Artemis command: ` org.apache.activemq.artemis.cli.commands.tools.PrintData`.
+It will reuse the configuration from the embedded broker to find the `bindings`, `messages` and `paging` directories.
+
+CLI to read the report:
+----
+attachment display --operation=/subsystem=messaging-activemq/server=default:print-data()
+----
+
+CLI to download a zipped version of the report:
+----
+attachment save --file=/home/ehsavoie/tmp/test.zip --operation=/subsystem=messaging-activemq/server=default:print-data(archive=true)
+----
+
+== Test Plan
+
+== Community Documentation
+////
+Generally a feature should have documentation as part of the PR to wildfly master, or as a follow up PR if the feature is in wildfly-core. In some cases though the documentation belongs more in a component, or does not need any documentation. Indicate which of these will happen.
+////
```

Comment:
@ehsavoie This section needs completing. I think it needs coverage in the regular docs (not just management API description).